### PR TITLE
palindrome-products: Add test template

### LINKED ISSUE #1964: palindrome-products: add test template
## Task
Create a test template file for `palindrome-products`

### Details
1. If this is your first time contributing to this repository, welcome!
    - Contributing to a repository on GitHub requires a [fork](https://help.github.com/en/articles/fork-a-repo)
1. Create a branch on your local repository called `palindrome-products-add-test-template`
    - Run: `git checkout -b palindrome-products-add-test-template`
1. Clone problem-specifications repository; this is used to populate the template
    - Run: `git clone https://github.com/exercism/problem-specifications ../problem-specifications`
1. Create `exercises/palindrome-products/.meta/template.j2`
    - See [Generator Documentation] for more information
1. Generate test file from new template
    - Run: `bin/generate_tests.py palindrome-products`

At this point, if the generator fails, or you are otherwise stuck,
please feel free to reach out! It is perfectly acceptable to submit
an incomplete PR in order to get feedback or assistance.

### Resources
- [Contributing Guide]
- [Generator Documentation]
- [Closing issues using keywords]

[Contributing Guide]: https://github.com/exercism/python/blob/master/docs/CONTRIBUTING.md
[Generator Documentation]: https://github.com/exercism/python/blob/master/docs/GENERATOR.md
[Closing issues using keywords]: https://help.github.com/en/articles/closing-issues-using-keywords

### PR DESCRIPTION
**Pull request by pywkm**
Closes #1964

### CODE REVIEW

File: exercises/palindrome-products/.meta/template.j2 [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+Notes regarding the implementation of smallest and largest:
+
+Both functions must take two keyword arguments:
+    max_factor -- int
+    min_factor -- int, default 0
+
+Their return value must be a tuple (value, factors) where value is the
+palindrome itself, and factors is an iterable containing both factors of the
+palindrome in arbitrary order.
+"""
```
Comment (cmccandless) [inline exercises/palindrome-products/.meta/template.j2]: This comment should be moved to `.meta/hints.md`, and the README will need to be [regenerated](https://github.com/exercism/python/blob/master/CONTRIBUTING.md#generating-exercise-readmes).

File: exercises/palindrome-products/.meta/template.j2 [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    {% for case in cases -%}
+    {{ test_case(case) }}
+    {% endfor -%}    
+
+    {{ macros.utility() }}
+
+    def assertFactorsEqual(self, actual, expected):
+        self.assertEqual(set(map(frozenset, actual)), set(map(frozenset, expected)))
+
+{{ macros.footer() }}
```
Comment (cmccandless) [inline exercises/palindrome-products/.meta/template.j2]:
```suggestion
{{ macros.footer(has_error_case) }}
```

File: exercises/palindrome-products/.meta/template.j2 [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+{% endmacro %}
+
+{{ macros.header() }}
+
+class {{ exercise | camel_case }}Test(unittest.TestCase):
+    {% for case in cases -%}
+    {{ test_case(case) }}
+    {% endfor -%}    
+
+    {{ macros.utility() }}
```
Comment (cmccandless) [inline exercises/palindrome-products/.meta/template.j2]:
```suggestion
```

File: exercises/palindrome-products/.meta/template.j2 (hunk 2 above)
Comment (pywkm) [reply exercises/palindrome-products/.meta/template.j2]: @cmccandless 

Yeah, I saw that parameter for footer, I even started with that, then changed to what you see now. Thing is, in the proposed way definition of method `assertFactorsEqual` doesn't land under section `# Utility functions` - it's above it. I wanted to make as little changes to the original `palindrome_products_test.py` file as possible.

If you're ok with this change, I am also, and will fix it.

Comment (cmccandless) [reply exercises/palindrome-products/.meta/template.j2]: Moving the exercise-specific utility function above the global one(s) is fine.

Comment (cmccandless) [reply exercises/palindrome-products/.meta/template.j2]: See #2084 ; providing the explicit argument is no longer required.

Comment (pywkm) [reply exercises/palindrome-products/.meta/template.j2]: Ok, changed.
If you don't mind I also fixed here the humming test template so it doesn't render utilities twice - spotted in your PR #2084

**Comment by pywkm** [comment]
Done, commit removed. :)

**Pull request merged**